cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1541 @@
+	flag.StringVar(&trelloCardID, "cardid", "", "sets the card ID for Trello board interaction")
+	flag.StringVar(&trelloListID, "listid", "", "sets the list ID for Trello board interaction")
+	flag.StringVar(&trelloBoardID, "boardid", "", "sets the board ID for Trello board interaction")
+	flag.StringVar(&trelloBoardName, "boardname", "", "sets the board name for Trello board interaction")
+	flag.StringVar(&exchangeName, "exchangename", "", "sets the exchangeName for the new exchange")
+	flag.StringVar(&checkType, "checktype", "", "sets the checkType for the new exchange")
+	flag.StringVar(&tokenData, "tokendata", "", "sets the tokenData for adding a new exchange")
+	flag.StringVar(&key, "key", "", "sets the key for adding a new exchange")
+	flag.StringVar(&val, "val", "", "sets the val for adding a new exchange")
+	flag.StringVar(&tokenDataEnd, "tokendataend", "", "sets the tokenDataEnd for adding a new exchange")
+	flag.StringVar(&textTokenData, "texttokendata", "", "sets the textTokenData for adding a new exchange")
+	flag.StringVar(&regExp, "regexp", "", "sets the regExp for adding a new exchange")
+	flag.StringVar(&dateFormat, "dateformat", "", "sets the dateFormat for adding a new exchange")
+	flag.StringVar(&path, "path", "", "sets the path for adding a new exchange")
+	flag.BoolVar(&add, "add", false, "used as a trigger to add a new exchange from command line")
+	flag.BoolVar(&verbose, "verbose", false, "increases logging verbosity for API Update Checker")
+	flag.BoolVar(&create, "create", false, "specifies whether to automatically create trello list, card and checklist in a given board")
+	flag.BoolVar(&firstRun, "firstrun", false, "specifies whether its the first time using the application on trello so all the exchanges are added to trello checklist")
+	flag.Parse()
+	var err error
+	c := log.GenDefaultSettings()
+	log.GlobalLogConfig = &c
+	log.SetupGlobalLogger()
+	configData, err = readFileData(jsonFile)
+	if err != nil {
+		log.Error(log.Global, err)
+		os.Exit(1)
+	}
+	testConfigData, err = readFileData(testJSONFile)
+	if err != nil {
+		log.Error(log.Global, err)
+		os.Exit(1)
+	}
+	usageData = testConfigData
+	if canUpdateTrello() {
+		usageData = configData
+	}
+	if add {
+		switch checkType {
+		case github:
+			var data GithubData
+			data.Repo = path
+			err = addExch(exchangeName, checkType, data, false)
+			if err != nil {
+				log.Error(log.Global, err)
+				os.Exit(1)
+			}
+		case htmlScrape:
+			var data HTMLScrapingData
+			data.TokenData = tokenData
+			data.Key = key
+			data.Val = val
+			data.TokenDataEnd = tokenDataEnd
+			data.TextTokenData = textTokenData
+			data.DateFormat = dateFormat
+			data.RegExp = regExp
+			data.Path = path
+			err = addExch(exchangeName, checkType, data, false)
+			if err != nil {
+				log.Error(log.Global, err)
+				os.Exit(1)
+			}
+		}
+	}
+	var a string
+	if canUpdateTrello() || create {
+		if trelloBoardName != "" {
```

Comment:
Should setAuthVars here if a person has only set there api key, token and board name, so that it gets set here